schain/schainwriter.go

```diff
@@ -57,13 +57,26 @@ func (w SChainWriter) Write(req *openrtb2.BidRequest, bidder string) {
 		sourceCopy := *req.Source
 		req.Source = &sourceCopy
 	}
-	schain := openrtb_ext.ExtRequestPrebidSChain{
-		SChain: *selectedSChain,
+
+	schain := openrtb_ext.ExtRequestPrebidSChain{}
+
+	if selectedSChain != nil {
+		schain.SChain = *selectedSChain
 	}
+
+	if hostSChainInfo != "" {
+		var hostSChainNode *openrtb_ext.ExtRequestPrebidSChainSChainNode
+		if err := json.Unmarshal([]byte(hostSChainInfo), &hostSChainNode); err != nil {
```

Comment:
I see we are unmarshalling the schain node because it was defined as a string. This leads me to believe it would be better if we defined the schain in the config as a struct. Alternatively we could allow the host to still define it as a string and unmarshal it into a struct at startup which would then be used here so we aren't performing an inefficient unmarshal on every request. I tend to lean though towards allowing the host to define it as a struct in their config. I think it results in leaner config code and is easier for the host to write. @mansinahar thoughts?

Reply:
Just to keep this updated, we talked during team time a couple days ago, and the type of the host schain node has now been reverted from string to be type `*openrtb_ext.ExtRequestPrebidSChainSChainNode`